Robinhood/Robinhood.py

```diff
@@ -100,32 +102,44 @@ def login(self,
 
         """
 
-        self.username = username
-        self.password = password
-        payload = {
-            'password': self.password,
-            'username': self.username
-        }
 
         if mfa_code:
-            payload['mfa_code'] = mfa_code
-
+            fields = { 'client_id' : 'c82SH0WZOsabOXGP2sxqcj34FxkvfnWRZBKlBjFS',
+                        'expires_in' : 86400,
+                        'grant_type': 'password',
+                        'password' : self.password,
+                        'scope' : 'internal',
+                        'username' : self.username,
+                        'mfa_code': self.mfa_code,
+                       }
+        else:
+            fields = {'client_id': 'c82SH0WZOsabOXGP2sxqcj34FxkvfnWRZBKlBjFS',
+                      'expires_in': 86400,
+                      'grant_type': 'password',
+                      'password': password,
+                      'scope': 'internal',
+                      'username': username,
+                       }
         try:
-            res = self.session.post(endpoints.login(), data=payload, timeout=15)
-            res.raise_for_status()
-            data = res.json()
-        except requests.exceptions.HTTPError:
-            raise RH_exception.LoginFailed()
-
-        if 'mfa_required' in data.keys():           # pragma: no cover
-            raise RH_exception.TwoFactorRequired()  # requires a second call to enable 2FA
-
-        if 'token' in data.keys():
-            self.auth_token = data['token']
-            self.headers['Authorization'] = 'Token ' + self.auth_token
-            return True
+            data = urllib.urlencode(fields) #py2
+        except:
+            #data = urllib.parse.urlencode(fields) #py3
+            data = fields
+
+        print(data)
+        res = self.session.post(endpoints.login(), data=data)
+        #res.raise_for_status()
+        res = res.json()
+        #print(data)
+        try:
+            #self.auth_token = res['token']
+            self.oauth_token = res['access_token']
+        except KeyError:
+            return res
 
-        return False
+        #self.headers['Authorization'] = 'Token '+self.auth_token
+        self.headers['Authorization'] = 'Bearer ' + self.oauth_token
```

Comment:
As I understand this `Bearer ` is not fixed but is also retrieved from auth endpoint as a token type